openpype/hosts/maya/plugins/publish/extract_xgen.py

```diff
@@ -0,0 +1,139 @@
+import os
+import copy
+
+from maya import cmds
+import pymel.core as pc
+import xgenm
+
+from openpype.pipeline import publish
+from openpype.hosts.maya.api.lib import maintained_selection, attribute_values
+from openpype.lib import StringTemplate
+
+
+class ExtractXgenCache(publish.Extractor):
+    """Extract Xgen"""
+
+    label = "Extract Xgen"
+    hosts = ["maya"]
+    families = ["xgen"]
+    scene_type = "ma"
+
+    def process(self, instance):
+        if "representations" not in instance.data:
+            instance.data["representations"] = []
+
+        staging_dir = self.staging_dir(instance)
+        maya_filename = "{}.{}".format(instance.data["name"], self.scene_type)
+        maya_filepath = os.path.join(staging_dir, maya_filename)
+
+        # Get published xgen file name.
+        template_data = copy.deepcopy(instance.data["anatomyData"])
+        template_data.update({"ext": "xgen"})
+        templates = instance.context.data["anatomy"].templates["publish"]
+        xgen_filename = StringTemplate(templates["file"]).format(template_data)
+        name = instance.data["xgenPalette"].replace(":", "__").replace("|", "")
+        xgen_filename = xgen_filename.replace(".xgen", "__" + name + ".xgen")
+
+        # Export xgen palette files.
+        xgen_path = os.path.join(staging_dir, xgen_filename).replace("\\", "/")
+        xgenm.exportPalette(
+            instance.data["xgenPalette"].replace("|", ""), xgen_path
+        )
+        self.log.info("Extracted to {}".format(xgen_path))
+
+        representation = {
+            "name": name,
+            "ext": "xgen",
+            "files": xgen_filename,
+            "stagingDir": staging_dir,
+        }
+        instance.data["representations"].append(representation)
+
+        # Collect nodes to export.
+        duplicate_nodes = []
+        for node, connections in instance.data["xgenConnections"].items():
+            transform_name = connections["transform"].split(".")[0]
+
+            # Duplicate_transform subd patch geometry.
+            duplicate_transform = pc.duplicate(transform_name)[0]
+            pc.parent(duplicate_transform, world=True)
+            duplicate_transform.name(stripNamespace=True)
+            duplicate_shape = pc.listRelatives(
+                duplicate_transform, shapes=True
+            )[0]
+
+            pc.connectAttr(
+                "{}.matrix".format(duplicate_transform),
+                "{}.transform".format(node),
+                force=True
+            )
+            pc.connectAttr(
+                "{}.worldMesh".format(duplicate_shape),
+                "{}.geometry".format(node),
+                force=True
+            )
```

Comment:
I would rewrite this code to not require `pymel`. Looking at how this is written you should be able to just replace it directly with `from maya import cmds` and replace `pc.` with `cmds.` except for the fact that it "reparents" and "duplicates" and thus the long name changes.
```suggestion
            # Duplicate_transform subd patch geometry.
            duplicate_transform = cmds.duplicate(transform_name)[0]
            duplicate_shape = cmds.listRelatives(
                duplicate_transform, 
                shapes=True, 
                fullPath=True
            )[0]

            cmds.connectAttr(
                "{}.matrix".format(duplicate_transform),
                "{}.transform".format(node),
                force=True
            )
            cmds.connectAttr(
                "{}.worldMesh".format(duplicate_shape),
                "{}.geometry".format(node),
                force=True
            )

            duplicate_transform = cmds.parent(duplicate_transform, world=True)[0]
```
I'd expect this to work.

Reply:
Any reason for not stripping the namespaces in your suggestion?

Reply:
nvm :)